Automattic/sensei
# Add interactivity to course steps

Resolves https://github.com/Automattic/sensei/issues/7446

## Proposed Changes
* Added interactivity for each of the steps of the Course tour
* Also adds the necessary structure and helpers to be reused by other tours

I have added tests for the functions and components. For the interactivity, this looks like a candidate for e2e testing. Otherwise, it doesn't look like it's possible to test the interactions reliably.

I'll create an issue for writing the e2e tests.
Also, ATM if someone clicks on a pagination step, we can't show the interactivity because tour-kit sends a wrong index to the callback. We're creating a PR in tour-kit library to fix it.

## Testing Instructions


1. Make sure your local env is set to development (to enable the feature flag)
2. Create a new Course or open an existing one in the editor (You'll need a new user in case you've closed the tour before, or you can remove the value of the usermeta `sensei_tours`)
3. Now go through each of the tour steps
4. Go both forward and backwards in the tour
5. Make sure the interactivity works as expected

https://github.com/Automattic/sensei/assets/6820724/e5335b72-166f-4b20-8b37-9f4f266bb8c0

## Pre-Merge Checklist
